python_modules/dagster/dagster/_daemon/controller.py

```diff
@@ -280,22 +279,6 @@ def check_daemon_loop(self) -> None:
                     self._workspace_process_context.refresh_workspace()
                     last_workspace_update_time = time.time()
 
-                if self._instance.daemon_skip_heartbeats_without_errors:
-                    # If we're skipping heartbeats without errors, we just check the threads.
-                    # If there's no errors, the daemons won't be writing heartbeats.
-                    continue
-
-                now = time.time()
-                # Give the daemon enough time to send an initial heartbeat before checking
-                if (
-                    (now - start_time) < 2 * self._heartbeat_interval_seconds
-                    or now - last_heartbeat_check_time < HEARTBEAT_CHECK_INTERVAL
-                ):
-                    continue
-
-                self.check_daemon_heartbeats()
-                last_heartbeat_check_time = time.time()
```

Comment:
should we warn after `HEARTBEAT_CHECK_INTERVAL` and die after `N` X `HEARTBEAT_CHECK_INTERVAL` ? 

while i agree the existing limit has proven problematic as the summary outlines
* warning when something "seems stuck" seems good
* theres probably some higher limit that matches "ya its probably stuck" even for assets/backfills? maybe?

Reply:
i can be convinced to just drop it, but seems worth some discussion